test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -4334,6 +4488,284 @@ def test_cgroup_default_config(self):
+        vmem_limit = os.path.join(mem_base, str(jid),
+                                  'memory.memsw.limit_in_bytes')
+        result = self.du.cat(hostname=self.mom.hostname, filename=vmem_limit,
+                             sudo=True)
+        vmem_limit_in_bytes = None
+        try:
+            vmem_limit_in_bytes = int(result['out'][0])
+        except Exception:
+            # None will be seen as a failure, nothing to do
+            pass
+        self.logger.info("job memsw limit: %d"
+                         % vmem_limit_in_bytes)
+        self.assertTrue(vmem_limit_in_bytes is not None,
+                        "Unable to read job memsw limit")
+
+        # Check results correspond to enforcement flags and job placement
+        swap_avail = vmem_avail_in_bytes - mem_avail_in_bytes
+        if enforce_flags[0] == 'true' and not exclhost:
+            self.assertTrue(mem_limit_in_bytes == 100 * 1024 * 1024,
+                            "Job mem limit is %d expected %d"
+                            % (mem_limit_in_bytes, 100 * 1024 * 1024))
+        else:
+            self.assertTrue(mem_avail_in_bytes == mem_limit_in_bytes,
+                            "job mem limit (%d) should be identical to "
+                            "total mem available (%d)"
+                            % (mem_limit_in_bytes, mem_avail_in_bytes))
+            self.logger.info("job mem limit is total mem available (%d)"
+                             % mem_avail_in_bytes)
+        if enforce_flags[1] == 'true' and not exclhost:
+            expected_vmem = (mem_limit_in_bytes
+                             + min(100 * 1024 * 1024, swap_avail))
+            self.assertTrue(vmem_limit_in_bytes == expected_vmem,
+                            "memsw limit: expected %d, got %d"
+                            % (expected_vmem, vmem_limit_in_bytes))
+            self.logger.info("job memsw limit is expected %d"
+                             % vmem_limit_in_bytes)
+        else:
+            if swap_avail:
+                self.assertTrue(vmem_avail_in_bytes == vmem_limit_in_bytes,
+                                "job memsw limit (%d) should be identical to "
+                                "total memsw available (%d)"
+                                % (vmem_limit_in_bytes, vmem_avail_in_bytes))
+                self.logger.info("job memsw limit is total memsw available "
+                                 " (%d)" % vmem_avail_in_bytes)
+            else:
+                self.assertTrue(mem_limit_in_bytes == vmem_limit_in_bytes,
+                                "no swap, mem (%d) and vmem (%d) limits "
+                                "should be identical but are not"
+                                % (mem_limit_in_bytes, vmem_limit_in_bytes))
+                self.logger.info("no swap: job memsw limit is job mem limit")
+
+    def test_cgroup_enforce_default_tf(self):
+        self.test_cgroup_enforce_default(enforce_flags=('true', 'false'))
```

Comment:
Please add a docstring.  Especially to clarify enforce_flags and which setting is being set to what without having to find the self.cfg16 and find it, i.e. memory.enforce_default=true and memsw.enforce_default=false

Reply:
Done

Reply:
Done